src/mcp/tools/project/list_apps.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint camelcase: 0 */
+
+import { z } from "zod";
+import { tool } from "../../tool.js";
+import { mcpError, toContent } from "../../util.js";
+import { AppPlatform, listFirebaseApps } from "../../../management/apps.js";
+
+const PLATFORM_MAP: Record<string, AppPlatform> = {
+  ios: AppPlatform.IOS,
+  android: AppPlatform.ANDROID,
+  web: AppPlatform.WEB,
+};
+
+export const list_apps = tool(
+  {
+    name: "list_apps",
+    description: "Retrieves apps registered in the current Firebase project.",
+    inputSchema: z.object({
+      platform: z
+        .enum(["ios", "android", "web"])
+        .nullish()
+        .describe("the specific platform to list (omit to list all platforms)"),
+    }),
+    annotations: {
+      title: "List Firebase Apps",
+      readOnlyHint: true,
+    },
+  },
+  async ({ platform }, { projectId }) => {
+    if (!projectId) return mcpError("No current project detected.");
+    const apps = await listFirebaseApps(projectId, PLATFORM_MAP[platform || ""] || AppPlatform.ANY);
```

Comment:
Nit: IIUC, this will return all apps if I call it with platform= 'foo'.  Is there something in MCP that enforces the enum? Or do we need to do so here?

Reply:
Yeah the MCP SDK enforces schema on the way in so the Zod enum should give us coverage here.